s3/s3_operations.cpp

```diff
@@ -245,19 +245,44 @@ namespace irods_s3 {
             }
         }
 
-
-        // if this is a replication and we're the destination, get the data size from the source dataObjInfo
+        // special treatment for replication - get data_size from source and
+        // number_of_threads from destination
         if (oprType == REPLICATE_DEST) {
+            bool found_data_size = false;
+            bool found_number_of_threads = false;
 
             for (int i = 0; i < NUM_L1_DESC; ++i) {
-
-                if (L1desc[i].inuseFlag && L1desc[i].dataObjInp && L1desc[i].dataObjInfo
-                       && L1desc[i].dataObjInp->objPath == file_obj->logical_path()
-                       && L1desc[i].dataObjInp->oprType == REPLICATE_SRC ) {
-
+                // get the data size from source dataObjInfo
+                if (!found_data_size && L1desc[i].inuseFlag && L1desc[i].dataObjInp && L1desc[i].dataObjInfo &&
+                    L1desc[i].dataObjInp->objPath == file_obj->logical_path() &&
+                    L1desc[i].dataObjInp->oprType == REPLICATE_SRC)
```

Comment:
Consider using references and/or pointers to the members of interest. For example:
```c++
const auto& l1d = L1desc[i];
const auto* dobj_input = l1d.dataObjInp;
const auto* dobj_info = l1d.dataObjInfo;
```

Reply:
done